lib/react/jsx.rb

```diff
@@ -14,11 +14,22 @@ module JSX
     # to provide your own transformer. It must implement:
     # - #initialize(options)
     # - #transform(code) => new code
+    #
+    # If you want to configure per-file transformation options with a proc, you may
+    # accept an optional second parameter
+    # - #transform(code, local_options = {}) => new code
     self.transformer_class = DEFAULT_TRANSFORMER
 
-    def self.transform(code)
+    def self.transform(code, input = {})
       self.transformer ||= transformer_class.new(transform_options)
-      self.transformer.transform(code)
+
+      if self.transformer.method(:transform).arity == 1
```

Comment:
I also prefer the not-checking-arity route. With the go-ahead to make a breaking change, I'll change the implementation to what you suggested!